### PR TITLE
Updates cocina-models to 0.64.0

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made?
Update cocina.

The substantive changes in this cocina release are:
* Add cocinaVersion attribute.
* Require purls for non-request descriptions.
* Require hasAgreements for APOs.

## How was this change tested?
Unit

### CODE REVIEW

**Comment by justinlittman** [comment]
Before:
```
Status (n=100000; not using Missing for success/different/error stats):
  Success:   96009 (96.088%)
  Different: 3909 (3.912%)
  Mapping error:     0 (0.0%)
  Create error:     0 (0.0%)
  Normalization error:     0 (0.0%)
  Missing from cache:     0 (0.0%)
  Unmapped:     0 (0.0%)
  Expected unmapped:     82 (0.082%)
  Bad cache:     0 (0.0%)
```


After:
```
Status (n=100000; not using Missing for success/different/error stats):
  Success:   96009 (96.088%)
  Different: 3909 (3.912%)
  Mapping error:     0 (0.0%)
  Create error:     0 (0.0%)
  Normalization error:     0 (0.0%)
  Missing from cache:     0 (0.0%)
  Unmapped:     0 (0.0%)
  Expected unmapped:     82 (0.082%)
  Bad cache:     0 (0.0%)
```

File: app/services/cocina/normalizers/mods_normalizer.rb [hunk 1]
```diff
@@ -18,7 +18,14 @@ def self.normalize(mods_ng_xml:, druid:, label:)
         ModsNormalizer.new(mods_ng_xml: mods_ng_xml, druid: druid, label: label).normalize
       end
 
-      def initialize(mods_ng_xml:, druid:, label:)
+      # @param [Nokogiri::Document] mods_ng_xml MODS to be normalized
+      # @param [String] druid
+      # @return [Nokogiri::Document] normalized MODS
+      def self.normalize_purl(mods_ng_xml:, druid:)
+        ModsNormalizer.new(mods_ng_xml: mods_ng_xml, druid: druid).normalize_purl
```
Comment (mjgiarlo) [inline app/services/cocina/normalizers/mods_normalizer.rb:25]: On lines 25 and 18, I believe you can drop `ModsNormalizer.`. Invoking `new()` should do the right thing.

File: app/services/cocina/normalizers/mods_normalizer.rb [hunk 2]
```diff
@@ -18,7 +18,14 @@ def self.normalize(mods_ng_xml:, druid:, label:)
...
+      end
+
+      def initialize(mods_ng_xml:, druid:, label: nil)
```
Comment (mjgiarlo) [inline app/services/cocina/normalizers/mods_normalizer.rb:28]: I notice that the `label` parameter now defaults to `nil` and it didn't before. Can you say more about this change?

File: app/services/cocina/to_fedora/descriptive/descriptive_writer.rb [hunk 3]
```diff
@@ -18,7 +18,7 @@ def self.write(xml:, descriptive:, druid:, id_generator: Descriptive::IdGenerato
           Subject.write(xml: xml, subjects: descriptive.subject, forms: descriptive.form, id_generator: id_generator)
           Descriptive::Event.write(xml: xml, events: descriptive.event, id_generator: id_generator)
           Identifier.write(xml: xml, identifiers: descriptive.identifier, id_generator: id_generator)
-          Descriptive::Access.write(xml: xml, access: descriptive.access, purl: descriptive.purl)
+          Descriptive::Access.write(xml: xml, access: descriptive.access, purl: descriptive.respond_to?(:purl) ? descriptive.purl : nil)
```
Comment (mjgiarlo) [inline app/services/cocina/to_fedora/descriptive/descriptive_writer.rb:21]: When would `descriptive` not respond to `:purl`? Presumably a `RequestDescription` instance?

**Review by mjgiarlo — COMMENTED** [review]
Have a few small questions but looks good otherwise!

File: app/services/cocina/to_fedora/descriptive/descriptive_writer.rb (hunk 3 above)
Comment (justinlittman) [reply app/services/cocina/to_fedora/descriptive/descriptive_writer.rb:21]: Correct.

File: app/services/cocina/normalizers/mods_normalizer.rb (hunk 2 above)
Comment (justinlittman) [reply app/services/cocina/normalizers/mods_normalizer.rb:28]: The new `normalize_purl` class method does not have/need the label and so does not provide it in initialization.

File: app/services/cocina/normalizers/mods_normalizer.rb (hunk 1 above)
Comment (justinlittman) [reply app/services/cocina/normalizers/mods_normalizer.rb]: Thanks! Fixed.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**